tools/validate-inventory.py

```
@@ -0,0 +1,178 @@
+
+    # Fail if effective_vars is completely empty
+    if not effective_vars:
+        print("\033[0;31m[FAIL] Inventory verification failed: No inventory variables could be loaded from either source.\033[0m")
+        sys.exit(1)
+
+    # Validate identities
+    violations = 0
+    for k, expected in required.items():
+        val = effective_vars.get(k)
+        if val is not None:
+            if val != expected:
```

Ruff (0.16.0)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)